operator/main.go

```diff
@@ -50,15 +50,40 @@ func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var tailingSidecarImage string
+	var configPath string
+	var config Config
+	var err error
+
 	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-	flag.StringVar(&tailingSidecarImage, "tailing-sidecar-image", "sumologic/tailing-sidecar:latest", "tailing sidecar image")
+	flag.StringVar(&tailingSidecarImage, "tailing-sidecar-image", "", "tailing sidecar image")
+	flag.StringVar(&configPath, "config", "", "Path to the configuration file")
 	flag.Parse()
 
 	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))
 
+	if configPath != "" {
+		config, err = ReadConfig(configPath)
+
+		if err != nil {
+			setupLog.Error(err, "unable to read configuration", "configPath", configPath)
+			os.Exit(1)
+		}
+	} else {
+		config = Config{}
+	}
+
+	if err := config.Validate(); err != nil {
+		setupLog.Error(err, "configuration error", "configPath", configPath)
+		os.Exit(1)
+	}
+
+	if tailingSidecarImage != "" {
+		config.Sidecar.Image = tailingSidecarImage
+	}
```

Comment:
I'd prefer if we had a default config factory in `config.go` instead of assembling it by hand here.

Reply:
What do you mean? Could you elaborate?

I added it here, because now they are 3 ways to set the image:

- default value (`config.go`)
- by configuration file (`config.go`)
- by flag (which I'm leaving due to backward compatibility)

I may add a function to config (`withSidecarImage`) and use it here, but it still be run from the same place

Reply:
Ah, my bad, I missed that this was a flag. In that case this is fine, though if we want to have more flags in the future, we should consider a more generic solution here.

Reply:
> In that case this is fine, though if we want to have more flags in the future, we should consider a more generic solution here.

I would like to drop flags in the future :D